src/torrent.cpp

```diff
@@ -5347,6 +5347,12 @@ namespace {
 #endif
 		}
 
+		if (!err)
+		{
+			if (alerts().should_post<file_prio_alert>())
+				alerts().emplace_alert<file_prio_alert>(get_handle());
```

Comment:
Can correct priorities be saved in resume data at this point?

Reply:
the whole reason to delay applying the file priorities is to *not* save the new file priorities in this case, since they failed to be applied. The problem with saving them here is that the next time the torrent is loaded, the location on disk for blocks will be based on the priorities, and blocks will go missing and there's a risk a full re-check is required .

As I've been thinking about this, I think it was a mistake to conflate the location on disk with the priorities, and those should probably be saved as separate entries in the resume data.

Reply:
Sorry, I thought this comment was on the error path, but it isn't. The answer to your question is "yes". and my previous comment applies to a few lines down, in `if (err) {`

Reply:
also, as I look at this. It doesn't look right. the `m_file_priority` member is updated before checking the error.

Reply:
oh, never mind. the disk thread will send back the new file priorities based on which ones succeeded.